test/plots/scatterPlotTests.ts

```diff
@@ -2,395 +2,379 @@
+        });
+        assert.deepEqual(closest.datum, dataset.data()[1], "correct datum has been retrieved");
+
+        svg.remove();
+      });
+
+      it("can retrieve Entities in a certain range", () => {
+        plot.renderTo(svg);
+
+        let entities = plot.entitiesIn({
+          min: xScale.scale(1),
+          max: xScale.scale(1)
+        }, {
+          min: yScale.scale(1),
+          max: yScale.scale(1)
+        });
+
+        assert.lengthOf(entities, 1, "only one Entity has been retrieved");
+        assert.deepEqual(entities[0].datum, { x: 1, y: 1 }, "correct datum has been retrieved");
+
+        svg.remove();
       });
 
-      it("setup is handled properly", () => {
-        assert.deepEqual(xScale.range(), [0, SVG_WIDTH], "xScale range was set by the renderer");
-        assert.deepEqual(yScale.range(), [SVG_HEIGHT, 0], "yScale range was set by the renderer");
-        circlePlot.selections().each(getCirclePlotVerifier());
-        assert.strictEqual(circlesInArea, 10, "10 circles were drawn");
+      it("does not return Entities whose center lies outside the range", () => {
+        plot.renderTo(svg);
+
+        let entities = plot.entitiesIn({
+          min: xScale.scale(1.001),
+          max: xScale.scale(1.001)
+        }, {
+          min: yScale.scale(1.001),
+          max: yScale.scale(1.001)
+        });
+
+        assert.lengthOf(entities, 0, "no Entities retrieved");
+
+        svg.remove();
+      });
+
+      it("can retrieve Entities in a certain bounds", () => {
+        plot.renderTo(svg);
+        let entities = plot.entitiesIn({
+          topLeft: {
+            x: xScale.scale(1),
+            y: yScale.scale(1)
+          },
+          bottomRight: {
+            x: xScale.scale(1),
+            y: yScale.scale(1)
+          }});
+
+        assert.lengthOf(entities, 1, "only one Entity has been retrieved");
+        assert.deepEqual(entities[0].datum, { x: 1, y: 1 }, "correct datum has been retrieved");
+
         svg.remove();
       });
 
-      it("rendering is idempotent", () => {
-        circlePlot.render();
-        circlePlot.render();
-        circlePlot.selections().each(getCirclePlotVerifier());
-        assert.strictEqual(circlesInArea, 10, "10 circles were drawn");
+      it("can retrieve Entities centered at a given Point", () => {
+        plot.renderTo(svg);
+
+        let entities = plot.entitiesAt({
+          x: xScale.scale(1),
+          y: yScale.scale(1)
+        });
+        assert.lengthOf(entities, 1, "only one Entity has been retrieved");
+        assert.deepEqual(entities[0].datum, { x: 1, y: 1 }, "correct datum has been retrieved");
+
         svg.remove();
       });
 
-      describe("after the scale has changed", () => {
-        beforeEach(() => {
-          xScale.domain([0, 3]);
-          yScale.domain([0, 9]);
+      it("determines whether an Entity contains a given Point by its position and size", () => {
+        let originalSize = 10;
```

Comment:
to suggest size will change and `originalRadius` won't work when the size changes